public/js/graph/react_node.js.jsx

```diff
@@ -203,60 +299,165 @@ var ReactNodes = React.createClass({
+            sat = this.props.parents.every(isAllTrue);     
+        } else if (this.state.logicalType === 'OR') {
+            sat = this.props.parents.some(isAllTrue);
+        }
+        return sat;
     },
     
-    handleMouseLeave: function(event) {
-        //code here
+    updateNode: function(svg) {
+        var newState =''
+        if (this.arePrereqsSatisfied(svg)) {
+            if (this.isSelected() || this.props.hybrid) {
+                newState='active';
+                this.setState({status: 'active'});
+            } else {
+                newState='takeable';
+                this.setState({status: 'takeable'});
+            }
+        } else {
+            if (this.isSelected() && !this.props.hybrid) {
+                newState='overridden';
+                this.setState({status: 'overridden'});
+            } else {
+                newState='inactive';
+                this.setState({status: 'inactive'});
+            }
+        }
+        
+        //setCookie(this.id, this.status);
+
+        // Always update children of hybrids
+        if (this.state.hybrid) {
+            $.each(this.props.childs, function(i, node) {
+                var currentNode = svg.refs['nodes'].refs[node] ? svg.refs['nodes'].refs[node] :
+                                                                 svg.refs['bools'].refs[node];
+                currentNode.setState({status: currentNode.state.status},
+                                     function(){this.updateNode(svg)}.bind(currentNode));
+            });
+            $.each(this.props.outEdges, function (i, edge) {
+                var currentEdge = svg.refs['edges'].refs[edge];
+                currentEdge.setState({status: currentEdge.state.status}, 
+                                     function(){this.updateEdge(svg)}.bind(currentEdge));
+            });
+        }
+
+    },
+    
+    turn: function(svg){
+        if (this.isSelected()) {  
+            if (this.props.parents.length == 0){
+                this.setState({status: 'takeable'}, function(){this.updateNode(svg)}.bind(this));
+            } else {
+                this.setState({status: 'inactive'}, function(){this.updateNode(svg)}.bind(this));
+            }
+        } else {
+            this.setState({status: 'active'}, function(){this.updateNode(svg)}.bind(this));
+        }
+                        
+        $.each(this.props.childs, function (i, node) {
+            var currentNode = svg.refs['nodes'].refs[node] ? svg.refs['nodes'].refs[node] :
+                                                             svg.refs['bools'].refs[node];
+            currentNode.setState({status: currentNode.state.status},
+                                 function(){this.updateNode(svg)}.bind(currentNode));
+        });
+        
+        $.each(this.props.outEdges, function (i, edge) {
+            var currentEdge = svg.refs['edges'].refs[edge];
+            currentEdge.setState({status: currentEdge.state.status}, 
```

Comment:
This line just screams like something is wrong, because it seems to be updating a state with its current value.